### PR TITLE
[refactor] Resolved several minor things on the code for #159

### PR DESCRIPTION
**Pull request by meronbrouwer**
@AbdelHajou 

After I checked out the code for #159 I found some minor things that I resolved in this pull request. Could you look at them and merge this PR, if you agree.

* Constructor chaining... chain them as you can see in these changes. This way there is only one path through all the constructors
* Make parameters `final`. It prevents changing them by accident in the method body
* Removed two unused setters.
* Added some extra generic types (most of these are reported by the problems view in IntelliJ)

### CODE REVIEW

File: src/main/java/com/github/hanyaeger/api/engine/entities/tilemap/EntityConfiguration.java [hunk 1]
```diff
@@ -4,36 +4,29 @@
 
 /**
  * A wrapper class that allows entities to be configured using a generic configuration object.
- *  * This configuration object can be passed to the entity's constructor for custom configuration.
+ * This configuration object can be passed to the entity's constructor for custom configuration.
+ *
  * @param <C> the type of the configuration object
  */
 class EntityConfiguration<C> {
```
Comment (AbdelHajou) [inline src/main/java/com/github/hanyaeger/api/engine/entities/tilemap/EntityConfiguration.java:11]: What do you think of using Lombok for data classes like these?

**Review by AbdelHajou — APPROVED** [review]
I should definitely check out the Problems view more, thank you for tagging me in this PR so I can see what I can improve on 👍

File: src/main/java/com/github/hanyaeger/api/engine/entities/tilemap/EntityConfiguration.java (hunk 1 above)
Comment (meronbrouwer) [reply src/main/java/com/github/hanyaeger/api/engine/entities/tilemap/EntityConfiguration.java:11]: Lombok nicely reduces the boileplate, but I'm actually waiting for Records to be included as a non-preview feature in one of the comming Java releases. You can read about it here: https://openjdk.java.net/jeps/359.

Its currently still in preview, but as soon as it leaves that stage, we can use it for all the data classes.

**Comment by meronbrouwer** [comment]
Since you're being so productive, I'll start using PR's for all my work. That will guaranty a better workflow and keep you up to speed on changes. When you lose time or interest, let me know and I'll stop assigning PR's to you.

g

**Comment by AbdelHajou** [comment]
@meronbrouwer Btw, I don't have permission to merge, so you'd have to do this yourself

**Comment by meronbrouwer** [comment]
Slowly he's requesting more power...

Just kidding, I'll change your priviliges.

**Comment by meronbrouwer** [comment]
@AbdelHajou You should have the correct permissions now (although I said that before).

Maybe you can start branching directly from master, in stead of your own fork, that way I can easily check out you feature branch and start playing with you new features.

**Comment by AbdelHajou** [comment]
@meronbrouwer I still can't merge this PR, maybe the orginazation permissions are overriding my permissions?

**Comment by meronbrouwer** [comment]
@AbdelHajou GitHub still says "pending invite", so I guess you have an invite in you inbox that you should accept

**Pull request merged**